ex1/dates_times.py

```diff
@@ -30,15 +30,57 @@
 Exercises for using the datetime and the calendar module
 '''
 
+from calendar import monthrange
+from datetime import date, time
+
 # Define a function named last_of_month that takes an argument dt of type date
 # and returns a date object representing the last day of the month dt was in.
 
+def last_of_month(dt):
+
+    '''
+    :param dt: date
+    :return: date object with the last day of the month dt was in
+    '''
+
+    return date(dt.year, dt.month, monthrange(dt.year, dt.month)[1])
+
 # Define a function named feed_the_gremlin which takes a time object as an
 # argument. It should return False between midnight and 6:30AM and True
 # otherwise.
 
+def feed_the_gremlin(t):
+    '''
+    :param t: time
+    :return: if t is between 0am and 0630am it returns False otherwise True
+    '''
+
+    return not t < time(6, 30)
+
 # Define a function named how_long that takes two datetime objects dt and ref
 # where ref is the reference datetime, calculates the difference between them and
 # returns the difference as a string formatted like:
 # "01 days, 01 minutes, 01 seconds until 2000-12-31 15:59:59"
 # If ref is before dt then use 'since' instead of 'until'
+
+def how_long(dt, ref):
+    '''
+    :param dt: datetime
+    :param ref: refference datetime
+    :return: the differnece between both imputs in days, minutes and seconds
+    '''
+
+    d = ref - dt
```

Comment:
Here you should have changed the direction of the comparison depending on which datetime is sooner.